### PR TITLE
feat: add email signup, email auth, add a passkey docs for Alchemy Signer

### PR DESCRIPTION
**Pull request by avasisht23**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
The focus of this PR is to enhance the Alchemy Signer functionality for email and passkey authentication and signup.

### Detailed summary
- Added functionality for email and passkey authentication and signup using the Alchemy Signer
- Updated documentation and examples for email and passkey authentication and signup
- Added components for email signup, email authentication, and passkey authentication

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: site/signers/alchemy-signer/email-auth.md
Comment (moldy530) [inline site/signers/alchemy-signer/email-auth.md:1]: email login and signup are handled by the same method, we should merge the two docs into the same doc and highlight that the method handles that for you

File: site/snippets/signers/alchemy/components/EmailAuth.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            iframeContainerId: "turnkey-iframe-container",
+          },
+        },
+      }),
+    []
+  );
+
+  // we are using react-query to handle loading states more easily,
+  // feel free to use whichever state management library you prefer
+  const authUser = useCallback(async () => {
```
Comment (moldy530) [inline site/snippets/signers/alchemy/components/EmailAuth.tsx:30]: this isn't quite right. you still have to initiate login with `signer.authenticate`

File: site/snippets/signers/alchemy/components/EmailSignup.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  const {
+    mutate: signup,
+    isLoading,
+    data: user,
+  } = useMutation({
+    mutationFn: () =>
+      // this will send an email to the user with a link to authenticate
+      signer.authenticate({
+        type: "email",
+        email: "USER_EMAIL",
```
Comment (moldy530) [inline site/snippets/signers/alchemy/components/EmailSignup.tsx:34]: let's add the redirect params as part of the example here as well

File: site/snippets/signers/alchemy/components/EmailAuth.tsx (hunk 1 above)
Comment (moldy530) [reply site/snippets/signers/alchemy/components/EmailAuth.tsx:30]: we should merge this component with the one below. we need to handle the `bundle` as well

File: site/snippets/signers/alchemy/components/EmailSignup.tsx (hunk 2 above)
Comment (moldy530) [reply site/snippets/signers/alchemy/components/EmailSignup.tsx:34]: you still need a listener for the bundle for signup and for login. the full flow is:
1. send user email
2. user clicks link in email
3. user hits website which reads the bundle from the url

File: site/snippets/signers/alchemy/email-signup.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { signer } from "./signer";
+
+signer.authenticate({
+  type: "email",
+  bundle: "USER_EMAIL",
```
Comment (moldy530) [inline site/snippets/signers/alchemy/email-signup.ts:5]: was this supposed to be `email: "USER_EMAIL"`?

let's also add the `redirectParams` field with a comment that it's optional and what it does

**Comment by avasisht23** [comment]
closing to refactor using hooks

**Pull request closed without merging**